### ISSUE TITLE
E0277 is not detected for method, which is implemented for another type

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#8275]
## Environment

* **IntelliJ Rust plugin version:** 0.4.161.4292-213
* **Rust toolchain version:** 1.59.0-nightly (0b42deacc 2021-12-09) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 (CL-213.5744.254)
* **Operating system:** macOS 11.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
E0277 is not detected for method, which is implemented for another type

## Steps to reproduce
There is a piece of code
https://doc.rust-lang.org/error-index.html#E0277
```rust
// here we declare the Foo trait with a bar method
trait Foo {
    fn bar(&self);
}

// we now declare a function which takes an object implementing the Foo trait
fn some_func<T: Foo>(foo: T) {
    foo.bar();
}

fn call_some_func() {
    // we now call the method with the i32 type, which doesn't implement
    // the Foo trait
    some_func(5i32); // error: the trait bound `i32 : Foo` is not satisfied
}
```

## Expected Results:
Error is highlighted and marked as E0277

## Actual Results:
There is no highlighting for the error
<img width="706" alt="Screenshot 2021-12-14 at 09 28 33" src="https://user-images.githubusercontent.com/1213122/145961174-17b32c42-a204-4eb9-b69b-8b7ede6c2e49.png">

Also, there is no E0277 for the following example, but anyway it is highlighted as error
```rust
fn some_func2<T>(foo: T) {
    println!("{:?}", foo); // error: the trait `core::fmt::Debug` is not
    //        implemented for the type `T`
}
```
<img width="605" alt="Screenshot 2021-12-14 at 09 31 09" src="https://user-images.githubusercontent.com/1213122/145961556-93625131-9d40-427d-843b-5d6b0d0c7952.png">
